File: Python/Algorithms/InsertionSort.py
def insertionSort(myarr):       # Worst Case Time Complexity = O(n^2) , S.C = O(1)
    for i in range(1,len(myarr)):
        for j in range(i-1,-1,-1):
            if myarr[j]<=myarr[j+1]: break
            myarr[j],myarr[j+1] = myarr[j+1],myarr[j]


arr = [9,2,7,4,6,5]
insertionSort(arr)
print(*arr)

File: Java/Algorithms/InsertionSort.java
package Java.Algorithms;

public class InsertionSort {

    public static void sort(int[] myarr){       // Worst Case Time Complexity = O(n^2) , S.C = O(1)
        for(int i = 1;i<myarr.length; i++){     
            for(int j = i-1;j>=0 ;j--){
                if (myarr[j]<=myarr[j+1]) break;

                int temp = myarr[j];
                myarr[j] = myarr[j+1];
                myarr[j+1] = temp;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {9,2,7,4,6,5};
        sort(arr);
        for(int val : arr){
            System.out.print(val+ " ");
        }
    }
}
